src/toolkit/matl_sell_policy.cc

```diff
@@ -285,7 +285,7 @@ void MatlSellPolicy::GetMatlTrades(
         trade_mat = mat;
       }
 
-      if (ignore_comp_) {
+      if (ignore_comp_ && it->request->target()->comp() != trade_mat->comp()) {
```

Comment:
Just to make sure.... do you want to ensure that these are not the same composition pointer? or not the same physical composition?  if the latter, you could use `compmath::AlmostEq()`.  That may be to "fuzzy" though, and maybe you do want to compare exact `Comp` pointers.

Reply:
Oh! Perfect, I didn't know about `compmath::AlmostEq()`. That's exactly what I'm looking for. I'll use that instead